### ISSUE TITLE
[feature] Print the conan version number during conan-create and conan-install

### ISSUE THREAD

**Issue by paulharris** [conan-io/conan#12718]
### What is your suggestion?

When upgrading recipes, it is important to know what version of conan the CI machines have, so I know when I can use features from the new conan version.

eg, when I can change `from conans.tools import stdcpp_library` to a conan v2 import.

Perhaps it is displayed in a PR somewhere, but I'd like to be able to look at a build log and see the version so there can be no doubt what was used, eg:

https://c3i.jfrog.io/artifactory/misc/logs/pr/13799/41-linux-gcc/libvpx/1.10.0/21a66396f9ee4dc0210d632a784f63f7f3e1d95b-build.txt

Just put it as the first line!
It would also help when people paste logs in slack or whatever...

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @paulharris 

I guess this is a feature request for conan-center-index, isn't it? Should I transfer the issue?

**Comment by memsharded** [comment]
I don't think that makes sense, for example in Conan 2.0 there are commands like: 
```
$ conan cache path pkg/version#revision
/path/to/cache/storage/of/this/recipe
```
This would become
```
$ conan cache path pkg/version#revision
Conan version 2.1.1
/path/to/cache/storage/of/this/recipe
```
And if we have to make the exception for some commands and not others, then it becomes a mess.
Furthermore, I don't know many other tools that does this, lets say, cmake, git, compilers... how many of them print their own version for every invocation of them? I think the standard way is having ``conan --version``, like any other tools.

**Comment by paulharris** [comment]
I also wouldn't want it for every conan command, eg conan search.  Those especially will feed into other tools, and extra diagnostic info is not desired...

But if you are printing out the [profile] and the dependency list of what is going to get built, then conan's version is also part of what is going into the mix and (I think) should displayed.

So just conan-create and conan-install, IMO.

**Comment by memsharded** [comment]
> So just conan-create and conan-install, IMO.

That seems very arbitrary. Whe ``conan info`` won't do it? It computes a full dependency graph ,it is almost an ``install`` command. Why not in the ``conan lock create`` that also computes a full dependency graph, also very similar? The reason why these commands show the profiles and the dependency lists is because those are variables, will vary for different invocations, different packages, different profiles or command line arguments. But the Conan version isn't a variable. 

Also why Conan will print its version, but CMake won't? Should Conan print the CMake version for every CMake usage? Because the CMake version can also be equally important?

For developers, knowing the version is just one ``conan --version`` away, as practically any other developer tool, I'd suggest to follow the same behavior. In my opinion your report clearly indicates that this is something that you want CI systems to display, and I would agree that it could be useful if C3I builds aren't showing this info, that it would be nice to have it printed in every build.

**Comment by paulharris** [comment]
I'm happy with that solution :)

**Comment by memsharded** [comment]
Transferring this to conan-center-index then :)
